app/lib/discourse_automation/triggers/user.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+DiscourseAutomation::Triggerable::USER = 'user'
+
+DiscourseAutomation::Triggerable.add(DiscourseAutomation::Triggerable::USER) do
+  on_update do |automation, metadata, previous_metadata|
+    list_id = metadata.dig('list_id', 'value')
+
+    if list_id
+
+      custom_field = "add_to_mailing_list_#{list_id}"
+
+      unless DiscoursePluginRegistry.self_editable_user_custom_fields.include?(custom_field)
```

Comment:
This looks a little unusual. It might be easier to use `register_editable_user_custom_field` here. 

More generally, though, I am wondering whether we can link the automation script to a user field created by the administrator. It feels problematic to have the Mailchimp-specific custom user fields in the automation plugin. They should either live in their own plugin or we can simplify the PR here and have the script use admin-created user fields. 

I'll move this discussion internally, for now.